level2.src.html

```diff
@@ -0,0 +1,120 @@
+</pre>
+
+<section>
+  <h2 id="intro">Introduction</h2>
+
+  <em>This section is not normative.</em>
+
+  When a user successfully loads a webpage from <code>example.com</code> over a
+  secure channel (HTTPS, for example), the user is guaranteed that no entity
+  between the user agent and <code>example.com</code> eavesdropped on or
+  tampered with the data transmitted between them. However, this guarantee is
+  weakened if the webpage loads subresources such as script or images over an
+  insecure connection. For example, an insecurely-loaded script can allow an
+  attacker to read or modify data on behalf of the user. An insecurely-loaded
+  image can allow an attacker to communicate incorrect information to the user
+  (e.g., a fabricated stock chart), mutate client-side state (e.g., set a
+  cookie), or induce the user to take an unintended action (e.g., changing the
+  label on a button). These requests are known as mixed content.
+
+  [[!MIXED-CONTENT]] details how a user agent can mitigate these risks by
+  blocking certain types of mixed content, and behaving more strictly in some
+  contexts. This specification updates and extends [[!MIXED-CONTENT]] to provide
+  better security and privacy guarantees while minimizing breakage.
+
+  The core idea of this specification is <i>mixed content
+  autoupgrading</i>. That is, mixed content that user agents are not already
+  blocking should be autoupgraded to a secure transport. If the request cannot
+  be autoupgraded, it will be blocked. Autoupgrading avoids loading insecure
+  resources on secure webpages, while minimizing the amount of developer effort
+  needed to avoid breakage.
+</section>
+
+<section>
+  <h2 id="terms">Key Concepts and Terminology</h2>
+
+  TODO
+</section>
+
+<section>
+  <h2 id="algorithms">Algorithms</h2>
+
+  <section>
+    <h3 id="upgrade-algorithm">Upgrade |request| to an <i lang="la">a priori</i>
+    authenticated URL as mixed content, if appropriate</h3>
+
+    TODO. This algorithm is called from Fetch before blocking requests as mixed content. It checks if the request is optionally-blockable mixed content, and if so rewrites the scheme to HTTPS.
+  </section>
+
+  <section>
+    <h3 id="modifications">Modifications to existing algorithms</h3>
+
+    TODO. Modify "Should fetching |request| be blocked as mixed content?" and "Should |response| to |request| be blocked as mixed content?" to ignore the optionally-blockable/blockable distinction. (All content that isn't autoupgraded should be blocked.)
+</section>
+
+<section>
+  <h2 id="integration">Integrations</h2>
+
+  TODO. Call the autoupgrading algorithm from Fetch.
+
+  TODO. Something about blocking insecure downloads from secure contexts (integrate with HTML).
+</section>
+
+<section>
+  <h2 id="obsolescences">Obsolescences</h2>
+
+  TODO. block-all-mixed-content is obsolete. Clarify the value of
+  upgrade-insecure-requests (upgrades blockable content as well as
```

Comment:
I was assuming we'd deprecate UIR too?

Reply:
Discussed a bit this morning -- depends on whether we want to autoupgrade blockable resources or just leave them as they are.